org.osgi.service.servlet/src/org/osgi/service/servlet/whiteboard/HttpWhiteboardConstants.java

```diff
@@ -571,5 +571,5 @@ private HttpWhiteboardConstants() {
 	 * The version of the implementation capability for the Servlet Whiteboard
 	 * specification
 	 */
-	public static final String	HTTP_WHITEBOARD_SPECIFICATION_VERSION				= "2.0";
+	public static final String	HTTP_WHITEBOARD_SPECIFICATION_VERSION				= "3.0";
```

Comment:
2.1 should be enough.
@cziegeler

Reply:
Right now I don't see a reason for an update of the spec at all. There is no new feature/functionality (unless I missed something)
If an update is needed for whatever reason, then yes 2.1 is enough
And I strongly encourage to avoid a major version change as this is a breaking change and makes it impossible (close to impossible) to provide an implementation support both versions. That's why we created a new package space when changing the spec to support Jakarta servlets. This allows to run javax servlets side-by-side with jakarta servlets.
The servlet API has breaking changes from 5 to 6 (which can mostly be neglected) but that's not OSGi's concern here.
I would also avoid tying a version to a servlet api version, meaning avoid to say that 2.0 can only be used with servlet 5 and 2.1 (if really required) can only be used with servlet 6.
Whiteboard 2.0 works happily with servlet API 6, no issue there. The Apache Felix Jetty12 Implementation passes all OSGi tests from R8.1 (and earlier): http service, http whiteboard, servlet whiteboard.